### ISSUE TITLE
bug:  incorrect shim set

### ISSUE THREAD

**Issue by ss-o** [z-shell/zi#87]
### Environment

linux-gnu|aarch64|5.8

### Current behavior

```zsh
~/.zi/plugins/sharkdp---fd/usr/share/bash-completion/completions/fd: 3: Syntax error: "(" unexpected (expecting "}")
```

```zsh
~/.zi/plugins/sharkdp---bat/usr/share/bash-completion/completions/bat: 16: Syntax error: "(" unexpected (expecting "fi")
```

```zsh
~/.zi/plugins/sharkdp---hyperfine/usr/share/bash-completion/completions/hyperfine: 3: Syntax error: "(" unexpected (expecting "}")
```

### Steps to reproduce

IN:

```zsh
zi ice as"command" from"gh-r" mv"fd* fd" sbin"**/fd(.exe|) -> fd"
zi light @sharkdp/fd
```

OUT:

```zsh
Downloading sharkdp/fd…
(Requesting `fd_8.3.1_arm64.deb'…)
######################################################################## 100.0%                                                                                                                                  
ziextract: Unpacking the files from: `fd_8.3.1_arm64.deb'…
ziextract: Successfully extracted and assigned +x chmod to the file: `usr/bin/fd'.
bin-gem-node annex: Created the fd shim and set +x on the fd binary
bin-gem-node annex: Created the fd shim and set +x on the fd binary
Installed 0 completions. They are stored in $INSTALLED_COMPS array.
Skipped installing 1 completions. They are stored in $SKIPPED_COMPS array.
```

---

IN:

```zsh
zi ice as"command" from"gh-r" mv"bat* bat" sbin"**/bat(.exe|) -> bat"
zi light @sharkdp/bat
```

OUT:

```zsh
Downloading sharkdp/bat…
(Requesting `bat_0.19.0_arm64.deb'…)
######################################################################## 100.0%                                                                                                                                  
ziextract: Unpacking the files from: `bat_0.19.0_arm64.deb'…
ziextract: Successfully extracted and assigned +x chmod to the file: `usr/bin/bat'.
bin-gem-node annex: Created the bat shim and set +x on the bat binary
bin-gem-node annex: Created the bat shim and set +x on the bat binary
Installed 0 completions. They are stored in $INSTALLED_COMPS array.
Skipped installing 1 completions. They are stored in $SKIPPED_COMPS array.
```

IN:

```zsh
zi ice as"command" from"gh-r" mv"hyperfine* hyperfine" sbin"**/hyperfine(.exe|) -> hyperfine"
zi light @sharkdp/hyperfine
```

OUT:
```zsh
Downloading sharkdp/hyperfine…
(Requesting `hyperfine_1.12.0_arm64.deb'…)
######################################################################## 100.0%                                                                                                                                  
ziextract: Unpacking the files from: `hyperfine_1.12.0_arm64.deb'…
ziextract: Successfully extracted and assigned +x chmod to the file: `usr/bin/hyperfine'.
bin-gem-node annex: Created the hyperfine shim and set +x on the hyperfine binary
bin-gem-node annex: Created the hyperfine shim and set +x on the hyperfine binary
Installed 0 completions. They are stored in $INSTALLED_COMPS array.
Skipped installing 1 completions. They are stored in $SKIPPED_COMPS array.
```

### Code snippet

The issue is that instead of setting shim to the `bin/...`, it is set to the completion:

```zsh
#!/usr/bin/env zsh

function fd {
  local bindir="/home/who-me?/.zi/plugins/sharkdp---fd/usr/share/bash-completion/completions"
  
  local -xU PATH="$bindir":"$PATH"
  
    "$bindir"/"fd" "$@"

}

fd "$@"
```

But same command call and same dir structure, hexyl installed correctly. 

```zsh
zi ice as"command" from"gh-r" mv"hexyl* hexyl" sbin"**/hexyl(.exe|) -> hexyl"
zi light @sharkdp/hexyl
```

For Linux could be fixed:

```zsh
zi ice as"command" from"gh-r" mv"bat* bat" sbin"**/bin/bat(.exe|) -> bat"
zi light @sharkdp/bat
```

However, need a closer look to pin when and why it happens as only arm devices are affected.

### Aditional information

These commands are set in a `meta-plugins` annex.

**Comment by anwalk** [comment]
Seems like for arm it picks `.deb` which unpacked to `/usr/bin/foo` and regular archives (`.zip`, `tar.gz`) extracted as `dir_of_archive_name/foo`. 

This picks correctly for `linux/darwin` - `x86_64/arm64`, haven't tested windows.

```zsh
zi ice as"command" from"gh-r" mv'**/fd fd' sbin'fd(.exe|) -> fd'; zi light @sharkdp/fd
```

**Comment by wicoop** [comment]
Then for Windows it's:

```zsh
zi ice as"program" from"gh-r" mv'**/fd(.exe|) -> fd' sbin'fd(.exe|)'; zi light @sharkdp/fd
```

Will them to the tests.

Cheers.

**Comment by ss-o** [comment]
Closing this issue as there is nothing that can be done in this repository and will be reflected at z-shell/z-a-meta-plugins. 

- The issue cause is the OS being Debian based, which extracts the `.deb` package which makes files extracted in different paths. I have solved this, but want to add extras to meta plugins and it will be reflected in ZI Wiki and meta plugins repository.

I'll add tacking of this issue to the ZI members board.
